solidity/dashboard/src/pages/rewards/TBTCRewardsPage.jsx

```diff
@@ -0,0 +1,187 @@
+import { ECDSARewardsHelper } from "../../utils/rewardsHelper"
+import { useWeb3Address } from "../../components/WithWeb3Context"
+import {
+  TokenAmountSkeleton,
+  DataTableSkeleton,
+} from "../../components/skeletons"
+import { gt } from "../../utils/arithmetics.utils"
+import { LoadingOverlay } from "../../components/Loadable"
+import EmptyStatePage from "./EmptyStatePage"
+// import { colors } from "../../constants/colors"
+
+const TBTCRewardsPage = () => {
+  const dispatch = useDispatch()
+  const yourAddress = useWeb3Address()
+  const currentIntervalEndOf = ECDSARewardsHelper.intervalEndOf(
+    ECDSARewardsHelper.currentInterval
+  ).unix()
+
+  const {
+    ecdsaAvailableRewardsFetching,
+    ecdsaAvailableRewardsBalance,
+    ecdsaAvailableRewards,
+    ecdsaRewardsHistory,
+  } = useSelector((state) => state.rewards)
+
+  useEffect(() => {
+    dispatch({
+      type: "rewards/ecdsa_fetch_rewards_data_request",
+      payload: yourAddress,
+    })
+  }, [dispatch, yourAddress])
+
+  const withdrawRewards = useCallback(
+    (submitButtonPromise) => {
+      dispatch({
+        type: "rewards/ecdsa_withdraw",
+        payload: ecdsaAvailableRewards,
+        meta: submitButtonPromise,
+      })
+    },
+    [dispatch, ecdsaAvailableRewards]
+  )
+
+  return (
+    <>
+      <RewardsOverview
+        balance={ecdsaAvailableRewardsBalance}
+        isBalanceFetching={ecdsaAvailableRewardsFetching}
+        withdrawRewards={withdrawRewards}
+      />
+      <section className="tile rewards-countdown">
+        <h2 className="h2--alt">
+          Next rewards release:&nbsp;
+          <Timer targetInUnix={currentIntervalEndOf} />
+        </h2>
+      </section>
+      <LoadingOverlay
+        isFetching={ecdsaAvailableRewardsFetching}
+        skeletonComponent={<DataTableSkeleton columns={4} />}
+      >
+        <section className="tile rewards-history">
+          <TBTCRewardsDataTable data={ecdsaRewardsHistory} />
+        </section>
+      </LoadingOverlay>
+    </>
+  )
+}
+
+const RewardsOverview = ({ balance, isBalanceFetching, withdrawRewards }) => {
+  return (
+    <section className="tile rewards__overview--tbtc">
+      <div className="rewards__overview__balance">
+        <h2 className="h2--alt text-grey-70 mb-1">tBTC Rewards</h2>
+        {isBalanceFetching ? (
+          <TokenAmountSkeleton />
+        ) : (
+          <TokenAmount amount={balance} currencySymbol="KEEP" />
+        )}
+      </div>
+      <div className="rewards__overview__period">
+        <h5 className="text-grey-70">current rewards period</h5>
+        <span className="rewards-period__date">
+          {ECDSARewardsHelper.periodOf(ECDSARewardsHelper.currentInterval)}
+        </span>
+        <div className="rewards-period__remaining-periods">
+          <Icons.Time width="16" height="16" className="time-icon--grey-30" />
+          <span>84 rewards periods remaining&nbsp;</span>
```

Comment:
I assume this is the number of intervals remaining. In that case we should calculate it. We are keeping the intervals number in `ECDSARewards` class but i don't think we have a method to calculate how many intervals remains until the end, so I would create one in there.

Reply:
pls see d8d138a but I addressed it in a different way than you suggested. What do you think?

Reply:
I think it's ok, we can keep it that way 👌